backend/tests/router_test.py

```diff
@@ -24,8 +26,9 @@ def test_get_agent_for_task_no_agent_found(mocker):
 
 def test_get_agent_for_task_agent_found(mocker):
     plan = {"agent_name": mock_agent_name}
-    mocker.patch("src.router.agents", mock_agents)
-    mocker.patch("src.router.llm", mock_model)
+    mocker.patch("src.router.get_llm", return_value=mock_model)

```

Comment:
Just so I understand, we're not actually using the get_llm function which needs the env file, instead we're instantiating a mock value for llm with 'mockllm' on line 8?

Reply:
Actually, this is a little confusing, I'll change. Yes, the test is mocking out `get_llm` so that whenever it is called during the test, the mock_model on line 8 will always be returned. The confuding part is that line 8 is actually using `get_llm` to get the mock model. It would make more sense if line 8 was:
```python
mock_model = MockLLM()
```

Reply:
Ah yes okay, thanks that makes sense